### ISSUE TITLE
Spurious reportUnknownParameterType for a known parameter type

### ISSUE THREAD

**Issue by lexicalunit** [microsoft/pyright#3319]
**Describe the bug**
```python
from authlib.integrations.django_client.apps import DjangoOAuth2App


def foo(app: DjangoOAuth2App) -> None:
    pass
```
Gives the error: `Type of parameter "app" is unknown (reportUnknownParameterType)`

**To Reproduce**
Steps to reproduce the behavior.

**Expected behavior**
The type is defined so the type of the parameter should not be unknown.

**VS Code extension or command-line**
Using the command line. Version 1.1.235.

**Additional context**
My `pyrightconfig.json` file includes the following settings:
```json
{
  "pythonVersion": "3.9",
  "reportUnknownParameterType": "error",
  "typeCheckingMode": "basic"
}
```

**Comment by hmc-cs-mdrissi** [comment]
Is authlib a typed library? Many python libraries are untyped. Untyped libraries are treated like Any/Unknowns and if that's case that error is correct. That error is fairly strict and if you use it either you use only typed libraries or you add your own type stubs for the parts of the libraries you use. I try to apply most of the strict rules, but this is one of the few I disable as I rely on a lot of untyped libraries and am slowly working on their stubs.

There is an easier version of that check called reportMissingParameterType that allows you to use unknown type annotations from untyped libraries.
